src/test/java/io/vertx/cassandra/CassandraServiceBase.java

```diff
@@ -15,40 +15,103 @@
  */
 package io.vertx.cassandra;
 
+import com.datastax.driver.core.BatchStatement;
+import com.datastax.driver.core.PreparedStatement;
+import io.vertx.core.CompositeFuture;
 import io.vertx.core.Future;
 import io.vertx.core.Vertx;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
 import org.apache.thrift.transport.TTransportException;
 import org.cassandraunit.utils.EmbeddedCassandraServerHelper;
 import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.util.concurrent.CountDownLatch;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static io.vertx.test.core.TestUtils.randomAlphaString;
 
 /**
  * A base class, which should be used for in all test where Cassandra service is required.
  */
+@RunWith(VertxUnitRunner.class)
 public class CassandraServiceBase {
 
+  private static final Logger log = LoggerFactory.getLogger(CassandraServiceBase.class);
+
   Vertx vertx = Vertx.vertx();
+
   static final String HOST = "localhost";
   static final int NATIVE_TRANSPORT_PORT = 9142;
+  private static final int BATCH_INSERT_SIZE = 1_000;
+  private static final int TIMES_TO_INSERT_BATCH = 100;
 
   @Before
-  public void before() throws InterruptedException, IOException, TTransportException {
+  public void before(TestContext context) throws IOException, TTransportException, InterruptedException {
     EmbeddedCassandraServerHelper.startEmbeddedCassandra();
 
-    CassandraClient cassandraClient = CassandraClient.create(
-      vertx,
-      new CassandraClientOptions()
-        .addContactPoint(HOST)
-        .setPort(NATIVE_TRANSPORT_PORT)
-    );
+    Future<Void> namesKeyspaceInitialized = initializeNamesKeyspace();
+    Future<Void> randomStringKeyspaceInitialized = initializeRandomStringKeyspace();
+
+    Async async = context.async();
+    CompositeFuture all = CompositeFuture.all(namesKeyspaceInitialized, randomStringKeyspaceInitialized);
+    all.setHandler(h -> {
+      if (h.failed()) {
+        log.error("Unable to initialize keyspaces", h.cause());
+        context.fail();
```

Comment:
Can you fail the context with the failure instead of logging it?

Reply:
Yes, just done